src/python.rs

```diff
@@ -7,12 +7,99 @@ use crate::gil::{self, GILGuard, GILPool};
 use crate::type_object::{PyTypeInfo, PyTypeObject};
 use crate::types::{PyAny, PyDict, PyModule, PyType};
 use crate::{ffi, AsPyPointer, FromPyPointer, IntoPyPointer, PyNativeType, PyObject, PyTryFrom};
-use std::ffi::CString;
+use std::ffi::{CStr, CString};
 use std::marker::PhantomData;
-use std::os::raw::c_int;
+use std::os::raw::{c_char, c_int};
 
 pub use gil::prepare_freethreaded_python;
 
+/// Represents the major, minor, and patch (if any) versions of this interpreter.
+///
+/// See [Python::version].
+#[derive(Debug)]
+pub struct PythonVersionInfo {
+    pub major: u8,
+    pub minor: u8,
+    pub patch: u8,
+    pub suffix: Option<String>,
```

Comment:
This could be `Option<&'a str>` if you want to avoid the malloc.

Reply:
Yeah; I was trying to keep lifetimes simple on the type but perhaps it's worth doing.